tests/test_pulse.py

```diff
@@ -68,3 +68,199 @@ def test_field_cos_with_chirp(make_pulse):
     # chirp breaks the even symmetry of the cos function
     assert cos[0] != cos[-1]
     assert cos[1] != cos[-2]
+
+
+@pytest.mark.parametrize("envelope_type", ENVELOPE_TYPES)
+def test_field_envelope(make_pulse, envelope_type):
+    fwhm = 15.0
+    t0 = 0
+    pulse = make_pulse(fwhm=fwhm, t0=t0, envelope_type=envelope_type)
+
+    t = np.array([-64, -16, -4.0, 0.0, 4.0, 16, 64])
+
+    envelope = pulse.calc_field_envelope(t)
+
+    s = snapshot(
+        {
+            "gauss": {
+                0: 1.0960549698939707e-11,
```

Comment:
We could save the reference values to a file, but I think it's quite useful to see them directly here.

Reply:
I like it that way. Fewer files with data for tests are better, having everything in one place, especially for testing internal functions. Thanks!